app/models/setting.rb

```diff
@@ -283,13 +278,18 @@ def self.load_defaults
     true
   end
 
+  def collection_method
+    "#{name}_collection".to_sym
+  end
+
+  def select_values
+    self.class.respond_to?(collection_method) ? self.class.send(collection_method) : nil
+  end
+
   def self.set(name, description, default, full_name = nil, value = nil, options = {})
     if options.has_key? :collection
-      SettingsHelper.module_eval do
-        define_method("#{name}_collection".to_sym) do
-          collection = options[:collection].call
-          collection.is_a?(Hash) ? collection : editable_select_optgroup(collection, :include_blank => options[:include_blank])
-        end
+      define_singleton_method("#{name}_collection".to_sym) do
```

Comment:
```suggestion
      define_singleton_method(collection_method_name) do
```

Reply:
I do not think this would work, we are inside class method and setting is not yet created.

Reply:
right, I've missed that nm me